packages/tools/addon/templates/components/cs-active-composition-panel.hbs

```diff
@@ -16,14 +16,16 @@
     enabled=editingEnabled
   }}
 
-
-  {{!-- TODO:
-      class=(concat "cs-toolbox-section" (if (has-errors fieldMark.model) 'invalid')
-      OR hasErrors=(has-errors model=fieldMark.model)
-  --}}
   {{#each fields key="id" as |fieldMark|}}
     {{#cs-collapsible-section
-      class="cs-toolbox-section"
+      class=(concat "cs-toolbox-section "
+        (if (cs-has-errors
+              fieldMark.model
+              fieldMark.model.content.isValid
```

Comment:
@ef4 The 2nd and 3rd parameters are only passed to the helper so that they trigger a re-computation of `cs-has-errors` when they change. That's unfortunate (and potentially confusing to the readers of the template/helper) but I cannot think of another way (that's not even worse, like creating a stateful helper and setting up an observer to trigger `recomputed` manually).